src/ImageSharp/Formats/Jpeg/JpegEncoderCore.cs

```diff
@@ -635,37 +296,47 @@ private void WriteDefineHuffmanTables(int componentCount)
                 markerlen += 1 + 16 + s.Values.Length;
             }
 
+            // TODO: this magic constant (array size) should be defined by HuffmanSpec class
+            // This is a one-time call which can be stackalloc'ed or allocated directly in memory as method local array
+            // Allocation here would be better for GC so it won't live for entire encoding process
+            // TODO: if this is allocated on the heap - pin it right here or following copy code will corrupt memory
```

Comment:
Main point of this is that previously this array was allocated for entire encoding run, including scan and everything else, not a big hit on gc but better to free sooner then later. Haven't though about pooling here, nice catch!
The only question is: use pool from provided image or from global configuration?

Reply:
But now it's stackalloced only for the `WriteDefineHuffmanTables` run isn't it? For a 179 byte array stackalloc is optimal, I still don't see why you consider to change it.

Reply:
Oh, I misunderstood you, it **was allocated** as class field for entire run before I moved everything to a separate class and stackallocted it :P
So that was a question: is stackalloc here good enough or should I change it to pooling/actual allocation.
I will remove TODOs and introduce constant for the size then.

Reply:
Yeah, the whole comment was about removing the TODO-s since they are misleading/unnecessary, stackalloc is more than good here.

Reply:
Actually, isn't it better to just write buffers like this?
```csharp
for (int i = 0; i < specs.Length; i++)
{
    this.outputStream.WriteByte(headers[i]);
    this.outputStream.Write(specs[i].Count);
    this.outputStream.Write(specs[i].Values);
}
```

No extra buffers, no extra constants, no memory copies and code is simplier by a lot.

Reply:
Yeah, those methods shouldn’t require anything more complicated. They’re called so few times.

Note to self: implement BufferedWriteStream.